Add page "SFP standard and ONT" with SFP with MAC vs SFP w/o MAC (#91)
hack-gpon/hack-gpon.github.io: _sfp/ont-wo-mac.md

---
title: SFP with PON MAC and w/o PON MAC
has_children: false
nav_order: 2
layout: default
---

PON technology unlike Ethernet are not P2P technologies but are one-to-many, and there are two device types: ONU (Optical Network Unit)/ONT (Optical Network Terminal) and OLT (Optical Line Terminal), both devices can be in SFP format[^tibit].

The OLT provides an integrated access box for Passive Optical Networks. OLTs are typically chassis with one or more line cards inside, and on each line card there is one or more PON transceiver, usually in SFP format. Each line card is connected to a secondary switch provides line card aggregation to the Ethernet uplinks. OLTs are often a mixture of Layer 2 and Layer 3 switching with traffic shaping on a per-customer, per-service basis[^tibit].

> The communication within the SFP PON transceiver is neither MMI nor Ethernet, outside [SPF standards](/sfp-standard.md), but rather it is an *equivalent electrical symbols of optical transmission* (which is simply the input/output of the [BOSA](/bosa-tosa-rosa.md)) that for simplicity's sake we would call **PON RAW communication** (also referred to as SFP w/o PON MAC) and all the PON management part is left to the card lines. Each equivalent electrical symbol of optical transmission is a separate dialect, distinct from other dialects, furthermore, as you can easily guess, this communication is not standard and is not within the signalling standards ([^sfprate],[^sfprate2],[^sfpplusstandard]) but is compliant with MSA with some part of [^sfpstandard],[^sfpplusstandard],[^sfpplusmi]. This requires extreme compatibility between ONT and transreciver. This design choice to use RAW communication is made for several reasons:

Reviewer: ```suggestion
The communication within the SFP PON transceiver is neither MMI nor Ethernet, outside [SFP standards](/sfp-standard.md), but rather it is an *equivalent electrical symbols of optical transmission* (which is simply the input/output of the [BOSA](/bosa-tosa-rosa.md)) that for simplicity's sake we call **PON RAW communication** (also referred to as SFP w/o PON MAC). All the PON management part is left to the line card itself. Each equivalent electrical symbol of optical transmission is a separate dialect, distinct from other dialects. Furthermore, as one can easily guess, this communication is not standard and is not within the signalling standards ([^sfprate],[^sfprate2],[^sfpplusstandard]) but it is compliant with some portions of  the MSA [^sfpstandard],[^sfpplusstandard],[^sfpplusmi]. This requires extreme compatibility between ONT and transreciver. This design choice is made for several reasons:
```

> - *size*: the size of an OLT w/o PON MAC is very similar to that of anMMI nor Ethernet transceiver, and the size of an OLT with the integrated PON MAC part far exceeds that of the SFP format

Reviewer: ```suggestion
- *size*: the size of an OLT w/o PON MAC is very similar to that of an MMI or Ethernet transceiver, and the size of an OLT with the integrated PON MAC far exceeds that of the standard SFP form factor
```

- *dissipative heating capacity*: the dissipative heating capacity of an OLT with PON MAC is higher than a normal transceiver, such as a 1 or 10 Gbps Ethernet link.
- *duplication*: there is a double `MAC` → `MMI` conversion (`MMI` → `MAC` → `PHY` → `MAC` → `OLT CPU`)

> - *repairability*: since lasers often have a shorter lifetime than other ICs, it is good to be able to change only the tranreciver

Reviewer: ```suggestion
- *repairability*: since lasers often have a shorter lifetime than other ICs, it is good to be able to change only the transceiver
```

> Despite this, there is a vendor that sells OLT SFP with PON MAC[^tibit]. The following pictures show an OLT SFP with PON MAC part and a transreciver without PON MAC. It is interesting to watch as it is much longer and requires an additional heatsink.

Reviewer: ```suggestion
Despite this, there is a vendor that sells OLT SFP with PON MAC[^tibit]. The following pictures show an OLT SFP with PON MAC part and a transreciver without PON MAC. It is interesting to see that the latter is much longer and requires an additional heatsink.
```

{% include image.html file="ont-wo-mac/tibit.png" alt="PON OLT with MAC" caption="PON OLT with MAC" %}
{% include image.html file="ont-wo-mac/huawei.png" alt="PON transceiver for OLT w/o MAC" caption="PON transceiver for OLT w/o MAC" %}

> Similarly, the same argument can be made for OLT SFPs, especially in 10E-PON and XGS-PON there are really a lot of transreceivers w/o PON MAC and few ONTs with PON MAC. In this case the reasons are similar to the previous ones. It is also evident that ONTs w/o PON MAC require a PON MAC part within the SFP cage that supports the relevant communication.

Reviewer: ```suggestion
Similarly, the same argument can be made for OLT SFPs, especially in 10E-PON and XGS-PON there are a lot of transreceivers w/o PON MAC and few ONTs with PON MAC. In this case the reasons are similar to the previous ones. It is also clear that ONTs w/o PON MAC require a PON MAC part within the end device that supports the relevant communication protocol.
```

{% include image.html file="ont-wo-mac/onu-with-mac.jpg" alt="Physical scheme of an ONT with MAC PON" caption="Physical scheme of an ONT with MAC PON" %}

```mermaid
graph TD
    F[fiber] --> A;

    subgraph SFP[SFP with MAC]
        B -->|Rx| A
        A[BOSA] -->|Tx| B(BOSA/LD Driver - Limiting Amplifier)
        
        B <--> C[PON MAC]
        C --> N[Nand Flash]
        C --> E[MCU - EEPROM]
    end;
    subgraph CS[Cage SFP]
    E --> |I2C| Controller;
    C --> |MMI on Tx - Rx| MAC;
    Controller[Controller I2C] --> Switch;
    MAC --> Switch;
    end;
```

{% include image.html file="ont-wo-mac/onu-wo-mac.jpg" alt="Physical scheme of an ONT w/o MAC PON" caption="Physical scheme of an ONT w/o MAC PON" %}

```mermaid
graph TD
    F[fiber] --> A;

    subgraph SFP[SFP w/o MAC]
        B -->|Rx| A
        A[BOSA] -->|Tx| B(BOSA/LD Driver - Limiting Amplifier)
        B <--> E[MCU - EEPROM]
    end;
    subgraph CS[Cage SFP]
    E --> |I2C| Controller;
    B --> |RAW PON on Tx - Rx| MAC[PON MAC];
    Controller[Controller I2C] --> Switch;
    MAC --> Switch;
    end;
```

# Why are there no ONT w/o MAC on Hack GPON?

> For utility reasons all SFPs w/o PON MAC and their cage SPF with PON MAC are not illustrated on Hack GPON as they are not moddable like ONT with MAC (they require 2 inter-compatible devices). 

Reviewer: ```suggestion
For utility reasons all SFPs w/o PON MAC are not illustrated on Hack GPON as they are not modifiable like ONT with MAC (they require two inter-compatible devices). 
```

> In particular, the SFP of AVM Fritz!Box 5530/5590 belongs in this category, and that the above-mentioned devices are not compatible with any SFP using MMI/Ethernet/Fibre Channel, while for example the FreeBox or IliadBox supports both ONT w/o PON MAC and some SFP with MAC.

Reviewer: ```suggestion
In particular, the SFP ONU of the AVM Fritz!Box 5530/5590 belongs in this category, and that the above-mentioned devices are not compatible with any other SFP using MMI/Ethernet/Fibre Channel, while for example the FreeBox or IliadBox supports both ONU w/o PON MAC and some SFP with MAC.
```
